### PR TITLE
kdoctor: init at 1.0.1

### PR DESCRIPTION
**Pull request by beisenherz**
Add kdoctor since it is a useful tool for working with kotlin multiplatform

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -15780,6 +15780,12 @@
       fingerprint = "B234 EFD4 2B42 FE81 EE4D  7627 F72C 4A88 7F9A 24CA";
     }];
   };
+  sironheart = {
```
Comment (AndersonTorres) [inline maintainers/maintainer-list.nix:15783]: Split this in a separate commit (no need for a new PR).

File: pkgs/development/tools/kdoctor/default.nix [hunk 2]
```diff
@@ -0,0 +1,27 @@
+{ lib, stdenv, fetchFromGitHub, installShellFiles }:
```
Comment (AndersonTorres) [inline pkgs/development/tools/kdoctor/default.nix:1]: One per line, to reduce future diffs:

```suggestion
{ lib
, stdenv
, fetchFromGitHub
, installShellFiles
}:
```

File: pkgs/development/tools/kdoctor/default.nix [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+stdenv.mkDerivation rec {
```
Comment (AndersonTorres) [inline pkgs/development/tools/kdoctor/default.nix:3]:
```suggestion
stdenv.mkDerivation (finalAttrs: {
```

File: pkgs/development/tools/kdoctor/default.nix [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+  version = "1.0.1";
+  pname = "kdoctor";
```
Comment (AndersonTorres) [inline pkgs/development/tools/kdoctor/default.nix:5]:
```suggestion
  pname = "kdoctor";
  version = "1.0.1";
```

File: pkgs/development/tools/kdoctor/default.nix [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+
+  src = fetchFromGitHub {
+    owner = "Kotlin";
+    repo = pname;
+    rev = "1.0.1";
```
Comment (AndersonTorres) [inline pkgs/development/tools/kdoctor/default.nix:10]:
```suggestion
    repo = "kdoctor";
    rev = finalAttrs.version;
```

File: pkgs/development/tools/kdoctor/default.nix [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+    longDescription = ''
+      KDoctor is a command-line tool that helps to set up the environment for
+      Kotlin Multiplatform Mobile app development.
+    '';
+    homepage = "https://github.com/Kotlin/kdoctor";
+    license = licenses.asl20;
+    platforms = platforms.darwin;
+    maintainers = with maintainers; [ sironheart ];
+  };
+}
```
Comment (AndersonTorres) [inline pkgs/development/tools/kdoctor/default.nix:27]:
```suggestion
})
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]
Are you sure this works? The original page says it requires Gradle...

**Comment by Sironheart** [comment]
Sorry, I did not realize the review would be made so fast 😅 I initially just wanted to see the build status of the CI, but I got my env working (because I'm stupid and didn't read the full documentation that took a while) and I for sure know now, that the build is working for me :)

File: pkgs/development/tools/kdoctor/default.nix [hunk 7]
```diff
@@ -0,0 +1,39 @@
+{ lib
+, stdenv
+, fetchurl
+, unzip
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  githubUser = "Kotlin";
```
Comment (AndersonTorres) [inline pkgs/development/tools/kdoctor/default.nix:8]: There is no need for parameterize everything. Use the value directly into the rest of the expression.

File: pkgs/development/tools/kdoctor/default.nix [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+  nativeBuildInputs = [ unzip ];
+
+  unpackPhase = ''
+      unzip $src -x META-INF/*
+    '';
+
+  installPhase = ''
+    mkdir -p $out/bin
+    install -m 0755 kdoctor $out/bin
+  '';
```
Comment (AndersonTorres) [inline pkgs/development/tools/kdoctor/default.nix:26]:
```suggestion
  unpackPhase = ''
    runHook preUnpack
      
    unzip $src -x META-INF/*

    runHook postUnpack

  '';

  installPhase = ''
    runHook preInstall

    mkdir -p $out/bin
    install -m 0755 kdoctor $out/bin

    runHook postInstall
  '';
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: pkgs/development/tools/kdoctor/default.nix [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+  src = fetchurl {
+    url = "https://github.com/Kotlin/${finalAttrs.pname}/releases/download/v${finalAttrs.version}/kdoctor_${finalAttrs.version}+92.zip";
+    hash = "sha256-rGK5U1JeXID9OcT8uzO3oO6f9MZgrW69LUHbEtj0SSQ=";
+  };
+
+  nativeBuildInputs = [ unzip ];
+
+  unpackPhase = ''
+    runHook preUnpack
+
```
Comment (AndersonTorres) [inline pkgs/development/tools/kdoctor/default.nix:20]: Consistency


```suggestion
```

File: pkgs/development/tools/kdoctor/default.nix [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+    unzip $src -x META-INF/*
+    runHook postUnpack
+  '';
+
+  installPhase = ''
+    runHook preInstall
+    mkdir -p $out/bin
+    install -m 0755 kdoctor $out/bin
```
Comment (AndersonTorres) [inline pkgs/development/tools/kdoctor/default.nix:28]:
```suggestion
    install -Dm755 kdoctor -t $out/bin/
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: maintainers/maintainer-list.nix (hunk 1 above)
Comment (AndersonTorres) [inline maintainers/maintainer-list.nix:15783]: Again:

Split this in a separate commit (no need for a new PR).

File: pkgs/development/tools/kdoctor/default.nix [hunk 11]
```diff
@@ -0,0 +1,43 @@
...
+  meta = with lib; {
+    description = "Environment analysis tool for Kotlin Multiplatform Mobile";
+    longDescription = ''
+      KDoctor is a command-line tool that helps to set up the environment for
+      Kotlin Multiplatform Mobile app development.
+    '';
+    homepage = "https://github.com/Kotlin/kdoctor";
+    license = licenses.asl20;
+    platforms = platforms.darwin;
+    maintainers = with maintainers; [ sironheart ];
```
Comment (AndersonTorres) [inline pkgs/development/tools/kdoctor/default.nix:41]:
```suggestion
    mainProgram = "kdoctor";
    maintainers = with maintainers; [ sironheart ];
    platforms = platforms.darwin;
    sourceProvenance = with sourceTypes; [ binaryNativeCode ];
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: maintainers/maintainer-list.nix (hunk 1 above)
Comment (Sironheart) [reply maintainers/maintainer-list.nix:15783]: Didn't I do this with the second commit? Hash d86a1052a5956d80f6ee735dcba3631177c8da1c

Comment (AndersonTorres) [reply maintainers/maintainer-list.nix:15783]: When I look the commit listing, I only see ` kdoctor: init at 1.0.1`. Four times.
Why?

Comment (Sironheart) [reply maintainers/maintainer-list.nix:15783]: I usually use the first line just the PR head and any additional lines for further information. I could squash/ reword the commits, if you want.

Comment (AndersonTorres) [reply maintainers/maintainer-list.nix:15783]: 1. If you want to add yourself to the maintainers file:
   1. **Do not open a pull request for it**
   2. Use the same pull request you are opening for including a package
   3. In case of need, Google "how to split a pull request into two"
   4. The specific commit for adding yourself to the maintainers file should follow the exact template below:

   ```
   maintainers: add <handle>
   ```

2. When adding a new package, we do not need to know how many times you reviewed and fixed it before inclusion. 
   1. Squash all those modifications
   2. The specific commit for adding a new package should follow the template below:

   ```
   <package name>: init at <version>

   <optional short description>
   ```

Comment (Sironheart) [reply maintainers/maintainer-list.nix:15783]: Fixed and reworded it. Sorry about that.

**Comment by AndersonTorres** [comment]
1. Invert the order of commits
2. Keep only the title of the maintainers commit

**Comment by Sironheart** [comment]
Done that

**Review by AndersonTorres — APPROVED** [review]

**Comment by AndersonTorres** [comment]
If no Darwin user complains in 3 days, I will merge this.

**Comment by Sironheart** [comment]
@AndersonTorres I believe I did so, via branch sync in my fork. Else I need to recheck

**Comment by AndersonTorres** [comment]
Nice.
